elastic_transport/client_utils.py

```diff
@@ -218,7 +219,11 @@ def url_to_node_config(
 
     headers = {}
     if parsed_url.auth:
-        username, _, password = parsed_url.auth.partition(":")
+        # `urllib3.util.url_parse` ensures `parsed_url` is correctly
+        # percent-encoded but does not percent-decode userinfo, so we have to
+        # do it ourselves to build the basic auth header correctly.
+        auth = urllib.parse.unquote(parsed_url.auth)
+        username, _, password = auth.partition(":")
```

Comment:
Would it not be (slightly) better to partition first, and then unquote? In the (extreme) case that the username contains an encoded colon (I can't imagine why, but it could) then this would partition in the wrong place. The actual separator we're looking for should never be encoded by definition, though.

Reply:
Thanks, fixed in https://github.com/elastic/elastic-transport-python/pull/143/commits/23f994608eb7f5e562c0f808eedd0bc0320510ab. (I meant "colon *in* userinfo username, not *is*.)